plugins/configs/api/listener_test.go

```diff
@@ -0,0 +1,102 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/influxdata/telegraf/agent"
+	"github.com/influxdata/telegraf/config"
+	"github.com/stretchr/testify/require"
+)
+
+func TestStatus(t *testing.T) {
```

Comment:
These endpoints don't enforce an HTTP method, in this test I changed `http.get` to `http.post` and sent an empty body and the test passed. We should have each of these endpoints only accept the valid methods.

Reply:
nice! I've updated those and tried changing a method in a test and got 405s